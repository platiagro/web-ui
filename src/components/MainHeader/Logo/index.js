// CORE LIBS
import React from 'react';
import PropTypes from 'prop-types';

/**
 * Logo.
 * This component is responsible for displaying logo.
 */
const Logo = ({ logoSrc, className, altText }) => (
  // logo div container
  <div className={className}>
    {/* logo image */}
    <img src={logoSrc} alt={altText} />
  </div>
);

// PROP TYPES
Logo.propTypes = {
  /** logo image src string */
  logoSrc: PropTypes.string.isRequired,
  /** logo css class string */
  className: PropTypes.string.isRequired,
  /** logo alt text string */
  altText: PropTypes.string.isRequired,
};

// EXPORT
export default Logo;
