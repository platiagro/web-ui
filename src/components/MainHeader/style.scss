.ant-layout-header {
  z-index: 1;
  padding: 0px;
}

.main-header-menu {
  width: 100%;
  line-height: 64px;
  background: linear-gradient(91.64deg, #3f6600 1.09%, #004e66 98.14%);
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);

  .ant-menu-item {
    a {
      color: #fff;
    }
  }
}

.ant-menu.ant-menu-dark .ant-menu-item-selected {
  background-color: #508300;
  font-weight: bold;
}

.logo {
  margin: 0 40px;
  float: left;
}

.logoDemo {
  margin: 0 40px;
  float: right;
}
