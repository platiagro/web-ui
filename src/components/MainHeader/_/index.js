// CORE LIBS
import React from 'react';

// UI LIBS
import { Layout } from 'antd';

// COMPONENTS
import Logo from '../Logo';
import MainMenu from '../MainMenu/Container';

// STYLES
import './style.scss';

// IMAGES
import logo from '../../../assets/logo-vazado.png';
import logoDemo from '../../../assets/logoDemo.png';

// LAYOUT COMPONENTS
const { Header } = Layout;

/**
 * Main Header.
 * This component is responsible for displaying the PlatIA logo, main menu and
 * Foragri demo logo.
 */
const MainHeader = () => (
  // header component
  <Header className='mainHeader'>
    {/* logo platiagro */}
    <Logo logoSrc={logo} className='logoPlatiagro' altText='PlatIAgro logo' />
    {/* main menu */}
    <MainMenu />
    {/* logo demo foragri */}
    <Logo
      logoSrc={logoDemo}
      className='logoDemo'
      altText='Demonstração ForAGRI'
    />
  </Header>
);

// EXPORT
export default MainHeader;
