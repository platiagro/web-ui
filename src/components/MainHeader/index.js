import React from 'react';

import { Link } from 'react-router-dom';

import { Layout, Menu } from 'antd';

import logo from '../../assets/logo.png';

import './style.scss';

const { Header } = Layout;

const MainHeader = () => (
  <Header>
    <div className='logo'>
      <img src={logo} alt='PlatIAgro' />
    </div>
    <Menu
      className='main-header-menu'
      theme='dark'
      mode='horizontal'
      defaultSelectedKeys={['2']}
    >
      <Menu.Item key='1'>
        <Link to='/'>In√≠cio</Link>
      </Menu.Item>
      <Menu.Item key='2'>
        <Link to='/projects'>Projetos</Link>
      </Menu.Item>
      <Menu.Item key='3'>
        <Link to='/implanted-models'>Modelos Implantados</Link>
      </Menu.Item>
    </Menu>
  </Header>
);

export default MainHeader;
