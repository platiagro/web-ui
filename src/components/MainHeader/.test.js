import React from 'react';

import { Link, Route, MemoryRouter } from 'react-router-dom';

import { shallow, mount } from 'enzyme';

import MainHeader from '.';

import { Menu } from 'antd';

describe('MainHeader component', () => {
  it('is expected render without crashing', () => {
    shallow(<MainHeader />);
  });

  it('is expected to contain a logo.', () => {
    const wrapper = shallow(<MainHeader />);

    expect(wrapper.find('.logo').exists()).toBeTruthy();
  });

  it('is expected to receive a mainRoutes property.', () => {
    const routes = [
      { path: '/', title: 'Root' },
      { path: '/projects', title: 'Projetos' },
    ];

    const wrapper = shallow(
      <MemoryRouter>
        <MainHeader mainRoutes={routes} />
      </MemoryRouter>
    );

    expect(wrapper.children().prop('mainRoutes')).toBe(routes);
  });

  it('is expected to have a child Menu', () => {
    const wrapper = shallow(<MainHeader />);

    expect(wrapper.children(Menu).exists()).toBeTruthy();
  });

  it('mainRoutes property is expected to map into Menu.Item components, children of Menu', () => {
    const routes = [
      { path: '/', title: 'Root' },
      { path: '/projects', title: 'Projetos' },
    ];

    const wrapper = shallow(<MainHeader mainRoutes={routes} />);

    expect(
      wrapper
        .children(Menu)
        .children(Menu.Item)
        .exists()
    ).toBeTruthy();
  });

  it(
    'Menu.Item component is expected to have a key with the path attribute ' +
      'of the route object, element of the mainRoutes vector',
    () => {
      const routes = [
        { path: '/', title: 'Root' },
        { path: '/projects', title: 'Projetos' },
      ];

      const wrapper = shallow(<MainHeader mainRoutes={routes} />);

      expect(
        wrapper
          .children(Menu)
          .children(Menu.Item)
          .get(0).key
      ).toBe(routes[0].path);
    }
  );

  it('Menu.Item component is expected to have a child Link', () => {
    const routes = [
      { path: '/', title: 'Root' },
      { path: '/projects', title: 'Projetos' },
    ];

    const wrapper = shallow(<MainHeader mainRoutes={routes} />);

    expect(
      wrapper
        .children(Menu)
        .children(Menu.Item)
        .children(Link)
        .exists()
    ).toBeTruthy();
  });

  it(
    'is expected to receive a location property (to receive this property ' +
      'the component must be routed `<Route component = {MainHeader} />`)',
    () => {
      const routes = [
        { path: '/', title: 'Root' },
        { path: '/projects', title: 'Projetos' },
      ];

      const wrapper = mount(
        <MemoryRouter>
          <Route component={MainHeader} />} />
        </MemoryRouter>
      );

      expect(wrapper.find(MainHeader).prop('location')).toBeDefined();
    }
  );

  it(
    'Menu selectedKeys props is expected to populate with the result of the' +
      'getCurrentRoute(location.pathname, mainRoutes) function call from the utils.js file.',
    () => {
      const routes = [
        { path: '/', title: 'Root' },
        { path: '/projects', title: 'Projetos' },
      ];

      const wrapper = mount(
        <MemoryRouter initialEntries={['/']} initialIndex={0}>
          <Route
            render={(props) => <MainHeader {...props} mainRoutes={routes} />}
          />
        </MemoryRouter>
      );

      expect(wrapper.find(Menu).prop('selectedKeys')).toEqual([routes[0].path]);
    }
  );

  it('Clicking on a menu item is expected to change a route', () => {
    const routes = [
      { path: '/', title: 'Root' },
      { path: '/projects', title: 'Projetos' },
    ];

    const wrapper = mount(
      <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <Route
          render={(props) => <MainHeader {...props} mainRoutes={routes} />}
        />
      </MemoryRouter>
    );

    wrapper
      .find(Link)
      .at(1)
      .simulate('click', { button: 0 });

    expect(wrapper.find(MainHeader).prop('location').pathname).toEqual(
      routes[1].path
    );
  });

  it('is expected render html correctly', () => {
    const wrapper = shallow(<MainHeader />);

    expect(wrapper).toMatchSnapshot();
  });
});
