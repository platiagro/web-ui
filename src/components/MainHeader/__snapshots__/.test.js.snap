// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader component renders html correctly 1`] = `
<Adapter>
  <div
    className="logo"
  >
    <img
      alt="PlatIAgro"
      src="logo.png"
    />
  </div>
  <Menu
    className="main-header-menu"
    defaultSelectedKeys={
      Array [
        "2",
      ]
    }
    mode="horizontal"
    theme="dark"
  >
    <MenuItem
      key="1"
    >
      <Link
        to="/"
      >
        In√≠cio
      </Link>
    </MenuItem>
    <MenuItem
      key="2"
    >
      <Link
        to="/projects"
      >
        Projetos
      </Link>
    </MenuItem>
    <MenuItem
      key="3"
    >
      <Link
        to="/implanted-models"
      >
        Modelos Implantados
      </Link>
    </MenuItem>
  </Menu>
</Adapter>
`;
